pkg/agent/openflow/network_policy.go

```diff
@@ -598,54 +695,161 @@ func (c *client) InstallPolicyRuleFlows(rule *types.PolicyRule) error {
 		if err := c.flowOperations.AddAll(actionFlows); err != nil {
 			return nil
 		}
+		// Add the action flows after the Openflow entries are installed on the OVS bridge successfully.
 		conj.actionFlows = actionFlows
 	}
+	c.conjMatchFlowLock.Lock()
+	defer c.conjMatchFlowLock.Unlock()
+
+	// Calculate the conjMatchFlowContext changes. The changed Openflow entries are included in the conjMatchFlowContext change.
+	ctxChanges := conj.getConjunctiveMatchFlowAddChange(c, rule)
+
+	// Send the changed Openflow entries to the OVS bridge, and then update the conjMatchFlowContext as the expected status.
+	if err := c.installConjunctiveMatchFlows(ctxChanges); err != nil {
+		return err
+	}
+	// Add the policyRuleConjunction into policyCache.
+	c.policyCache.Store(rule.ID, conj)
+	return nil
+}
+
+// installConjunctiveMatchFlows installs Openflow entries on the OVS bridge, and then updates the conjMatchFlowContext.
+// Release the lock in the conjMatchFlowContextChange no matter the Openflow entries are installed on the OVS
```

Comment:
What is the "lock in the conjMatchFlowContextChange"? The only lock I see is `conjMatchFlowLock` in the client.